packages/engine/src/avatar/functions/moveAvatar.ts

```diff
@@ -123,7 +122,9 @@ export function updateLocalAvatarPosition(additionalMovement?: Vector3) {
     return
   }
 
-  if (controller.movementEnabled && additionalMovement) desiredMovement.add(additionalMovement)
+  if (!controller.movementEnabled) return
```

Comment:
This will break XR locomotion. Wven if the controller is disabled, we need to run through the rest of this logic to ensure real-life movement is taken into account